Services/UserService.cs

```diff
@@ -29,55 +30,27 @@ public UserService(IOptions<DirectoryConfig> config)
                 throw new Exception("Invalid config");
             }
 
-            var insecure = opts.Insecure;
-            var directoryServiceURL = opts.ServiceUrl;
+            var logggerFactory = new NullLoggerFactory();
+            var options = new AsertoDirectoryOptions(opts.ServiceUrl, opts.APIKey, opts.TenantID, opts.Insecure);
 
-            var grpcChannelOptions = new GrpcChannelOptions { };
-
-            if (insecure)
-            {
-                var httpHandler = new HttpClientHandler
-                {
-                    ServerCertificateCustomValidationCallback =
-                        HttpClientHandler.DangerousAcceptAnyServerCertificateValidator,
-                };
-
-                grpcChannelOptions = new GrpcChannelOptions { HttpHandler = httpHandler };
-            }
-
-            var channel = GrpcChannel.ForAddress(
-                directoryServiceURL,
-                grpcChannelOptions);
-
-            this.directoryReaderClient = new ReaderClient(channel);
+            var optionsInt = Microsoft.Extensions.Options.Options.Create(options);
+            this.directoryClient = new DirectoryAPIClient(optionsInt, logggerFactory);
```

Comment:
You can omit `this` and just have `directoryClient = new DirectoryAPIClient(optionsInt, logggerFactory);`